src/utils/AddToPlaylistModal.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { useState } from "react";
+import { useVideoLib, dispatchTypeEnum } from "../context";
+
+export function AddToPlaylistModal({ videoId, toggleModal }) {
+  const { state, dispatch } = useVideoLib();
+
+  const [playlistName, setPlaylistName] = useState("");
+
+  function addNewPlaylistHandler() {
+    const newPlaylist = playlistName;
+    if (newPlaylist !== "") {
+      dispatch({ type: dispatchTypeEnum.ADD_NEW_PLAYLIST, payload: newPlaylist });
+    }
+    setPlaylistName("");
+  }
+
+  return (
+    <div className="modal-container">
+      <div class="modal">
+        <div class="modal-title">
+          <h3>Playlists</h3>
+        </div>
+        <div>
+          <input
+            value={playlistName}
+            onChange={(e) => setPlaylistName(e.target.value)}
+            placeholder="create new"
+          />
+          <button onClick={addNewPlaylistHandler} className="bttn">
+            +
+          </button>
+        </div>
+        <div class="modal-body">
+          {state.playlist.map((list) => (
+            <label>

```

Comment:
You might be getting a warning here about the list not having a key attribute. Might want to look into it. These warnings are suppressed in the deployed version, so I couldn't check it.